common/src/java/com/zimbra/common/calendar/ZoneInfo2iCalendar.java

```diff
@@ -397,8 +400,31 @@ private static Observance toObservanceComp(int hintYear, RuleLine rline, boolean
         }
     }
 
+    private static String replaceTzNameVariable(Time standardOffset, Time daylightOffset) {
+        String stdOffset = null;
+        String dlOffset = null;
+        stdOffset = new TzOffsetTo(new UtcOffset(getUtcOffset(standardOffset))).getValue();
+        stdOffset = stdOffset.replaceAll("00$", "");
+
+        if (daylightOffset != null) {
+            dlOffset = new TzOffsetTo(new UtcOffset(getUtcOffset(daylightOffset))).getValue();
+            dlOffset = dlOffset.replaceAll("00$", "");
+        }
+
+        if (dlOffset == null || stdOffset.equals(dlOffset)) {
+            return stdOffset;
+        } else {
+            return stdOffset + "/" + dlOffset;
+        }
+    }
+
     private static Standard toStandardComp(Time gmtOffset, String tznameFormat) {
         PropertyList props = new PropertyList();
+        if (tznameFormat != null && tznameFormat.equals("%z")) {
+            if (tznameFormat.equals("%z")) {
```

Comment:
Do we require this if. Above if statement already handle this!

Reply:
You're right. Thank you for pointing it out.